resources/speechdetector/src/main/java/org/restcomm/media/resource/speechdetector/SpeechDetector.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2017, Telestax Inc and individual contributors
+ * by the @authors tag. 
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.restcomm.media.resource.speechdetector;
+
+import org.restcomm.media.MediaSink;
+
+/**
+ * Components that detects user speech from a stream of incoming audio.
+ * 
+ * @author anikiforov
+ *
+ */
+public interface SpeechDetector extends MediaSink {
```

Comment:
Do not extends from MediaSink. Let's make this a decoupled, light, component.
Also, remove `startSpeechDetection` and `stopSpeechDetection` methods and replace them with a simple `detect(byte[] data, int offset, int len)` method or similar.

This style of start/stop methods, which require a certain order of invocation for the component to work correctly, is an anti-pattern. Let's avoid this at all costs.